app/scripts/controllers/app-state.js

```diff
@@ -107,6 +121,16 @@ export default class AppStateController extends EventEmitter {
     this._resetTimer()
   }
 
+  /**
+   * Sets the "switch to connected" alert as shown for the given origin
+   * @param {string} origin - The origin the alert has been shown for
+   */
+  setSwitchToConnectedAlertShown (origin) {
```

Comment:
After discussing this with @rachelcope , we've decided to set this alert to "shown" after the user sees the _other_ alert too (the "Unconnected account" alert seen after switching accounts in the popup). This is to prevent them from seeing two similar alerts in a row.

An example of this case is:

* The user opens the popup on a dapp, with a connected account selected
* The user switches to an unconnected account, sees the "unconnected account" alert, then dismisses it
* The user closes the popup
* The user opens the popup again

In that case, currently we'd show the "switch to account" alert. This is what we'd like to change, so that this alert is skipped in this specific scenario.

Reply:
This has been done in 6b320b70505425a342bb9268f1f180e2da79464e